### ISSUE TITLE
[TwigBridge] Unexpected deprecation notices for Twig\Extension::getName()

### ISSUE THREAD

**Issue by chalasr** [symfony/symfony#24522]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.4

The Debug component detects usages of deprecated methods since https://github.com/symfony/symfony/pull/23816.

Problem is that the TwigBridge's Twig extensions are all implementing `Twig\Extension::getName()` which is deprecated since Twig 1.26 (only via `@deprecated` phpdoc annotations to avoid unexpected deprecation notices, see https://github.com/twigphp/Twig/pull/2148).

As a result, using symfony/debug 3.4 + symfony/twig-bridge 3.4 + twig 1.x triggers the following deprecation:

> The "Twig_Extension::getName()" method is deprecated since 1.26 (to be removed in 2.0), not used anymore internally. You should not extend it from "Symfony\Bridge\Twig\Extension\AssetExtension".

Same for every Twig extension provided by TwigBridge (16)

**Comment by javiereguiluz** [comment]
It's not only about `getName()` method. I also see this deprecation when upgrading to 3.4:

```
The "Twig_Extension::initRuntime()" method is deprecated since 1.23 (to be
removed in 2.0), implement Twig_Extension_InitRuntimeInterface instead. You
should not extend it from "Symfony\Bridge\Twig\Extension\FormExtension".
```

**Comment by stof** [comment]
The Debug component now forbids keeping BC with older versions of a library (and also does not support the case where we want to move the method to a new interface, like Twig_Extension_InitRuntimeInterface)

**Comment by chalasr** [comment]
Fixed by #24921
